src/components/ShopingCart/ShopingCart.jsx

```diff
@@ -0,0 +1,198 @@
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfLectures: 4,
+    numOfSections: 8,
+    price: 100.0,
+    rating: 3,
+  },
+  {
+    image: course4,
+    author: "elzero",
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 120,
+    numOfLectures: 5,
+    numOfSections: 8,
+    price: 300.0,
+    rating: 2.5,
+  },
+  {
+    image: course5,
+    author: "Karim abdelazim",
+    title: "Data Structure",
+    category: "Programming",
+    numOfVideos: 50,
+    numOfLectures: 10,
+    price: 400.0,
+    rating: 5,
+  },
+  {
+    image: course2,
+    author: "Adel nsiem",
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfLectures: 4,
+    numOfSections: 8,
+    price: 300.0,
+    rating: 3.5,
+  },
+  {
+    image: course4,
+    author: "Adel nsiem",
+    title: "React Js",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfLectures: 4,
+    price: 50.0,
+    rating: 4.5,
+  },
+  {
+    image: course3,
+    author: "Adel nsiem",
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfSections: 8,
+    numOfLectures: 4,
+    price: 50.0,
+    rating: 4.5,
+  },
+  {
+    image: course1,
+    author: "Adel nsiem",
+    title: "The complete course of programming object for beginners",
+    category: "Programming",
+    numOfVideos: 150,
+    numOfLectures: 4,
+    numOfSections: 8,
+    price: 50.0,
+    rating: 4.5,
+  },
+];
+function ShopingCart() {
+  const [currentPage, setCurrentPage] = useState(1);
+  const productsPerPage = 4;
+  const getCurrentPage = (pageIndex, productsPerPage) => {
+    setCurrentPage(pageIndex);
+  };
+  const indexOfLastCourse = currentPage * productsPerPage;
+  const indexOfFirstCourse = indexOfLastCourse - productsPerPage;
+  const currentProducts = ShopingCartCourses.slice(
+    indexOfFirstCourse,
+    indexOfLastCourse
+  );
+  return (
+    <>
+      {/* Header Of Shopping Cart */}
+      <div className="flex justify-center place-content-center bg-stone-700 bg-opacity-8 p-[1.5rem] w-full">
+        <h1 className="text-white text-[1rem] md:text-[1.5rem] xl:text-[2rem] font-semibold tracking-wider">
+          Shopping Cart <FontAwesomeIcon icon={faShoppingCart} />
+        </h1>
+      </div>
+      {/* render courses */}
+      <div className="flex flex-col md:flex-row mx-4 mb-8">
+        <div className="ml-0 md:ml-10 xl:ml-32 mt-20 w-full md:w-7/12">
+          <div className="mb-3">
+            <h1 className="text-2xl font-bold">Courses</h1>
+          </div>
+          {ShopingCartCourses.slice(
+            (currentPage - 1) * productsPerPage,
+            currentPage * productsPerPage
+          ).map((item) => {
+            return (
+              <div
+                key={item}
```

Comment:
Here we should use the index for the "key" property because it would be duplicate values in the array (same item)

Reply:
for the second time sorry .